packages/ccip-svm/src/encoders/svmExtraArgsV1.ts

```diff
@@ -0,0 +1,131 @@
+import { address, getAddressEncoder, getAddressDecoder, getAddressFromPublicKey } from '@solana/kit'
+
+/**
+ * SVM Extra Args V1 tag: bytes4(keccak256("CCIP SVMExtraArgsV1"))
+ */
+export const SVM_EXTRA_ARGS_V1_TAG = 0x1f3b3aba
+
+export type Address = ReturnType<typeof address>
+
+export interface SVMExtraArgsV1 {
+    computeUnits?: number
+    accountIsWritableBitmap?: bigint
+    allowOutOfOrderExecution?: boolean
+    tokenReceiver?: Address
+    accounts?: Address[]
+}
+
+/**
+ * Creates properly encoded extraArgs buffer for SVM destinations
+ * @param options - SVM extra args configuration
+ * @returns Properly encoded extraArgs as Uint8Array
+ */
+export function encodeSVMExtraArgsV1({
+    computeUnits = 0,
+    accountIsWritableBitmap = 0n,
+    allowOutOfOrderExecution = true,
+    tokenReceiver,
+    accounts = []
+}: SVMExtraArgsV1): Uint8Array {
```

Comment:
Just had a thought:  To make this encodeFunction easier to use (in evm and SVM) should it return hex rather than Uint8Array?  In the tests for example we have convert the value of `expected` into uint8Array
From a devx perspective will most people recognise hex more?

we could offer a utility (if needed by the SDK) to convert hex to Uint8Array but is there a better devx if encode/decode etc work with 0x-prefixed hexes to make it more familiar to users (including where uncaught exceptions arise etc?)